### PR TITLE
[web]: Simplifies circuits.web.HTTP properties by removing useless hasattr/getattr checks

### PR DESCRIPTION
**Pull request by prologic**

### CODE REVIEW

File: circuits/web/http.py [hunk 1]
```diff
@@ -65,20 +65,14 @@ def protocol(self):
 
     @property
     def scheme(self):
-        if not hasattr(self, "_server"):
-            return
         return "https" if self._server.secure else "http"
 
     @property
     def base(self):
-        if getattr(self, "uri", None) is None:
```
Comment (apollo13) [inline circuits/web/http.py]: If `self.uri` is `None` (ie before the `ready` handler ran), the line below `self.uri.utf8()` would error out if. I guess you still need to check if `self.uri` is `None` but you can remove the `getattr`

Comment (prologic) [inline circuits/web/http.py]: Good catch. Fixed.

**Comment by prologic** [comment]
Review please :)

File: circuits/web/http.py [hunk 2]
```diff
@@ -65,20 +65,16 @@ def protocol(self):
...
-            return
+        if self.uri is None:
+            return ""
```
Comment (spaceone) [inline circuits/web/http.py]: Wrong... return b""!

Comment (apollo13) [inline circuits/web/http.py]: Then the code below is also wrong since it explicitelly decodes ;)

Comment (spaceone) [inline circuits/web/http.py]: oh yes... then everything is fine.

**Pull request merged**